### PR TITLE
lib/patternfly: Remove :focus-visible outline from some elements

### PR DESCRIPTION
**Pull request by mvollmer**
- [ ] https://github.com/patternfly/patternfly/issues/4121

### CODE REVIEW

File: pkg/lib/patternfly/patternfly-4-overrides.scss [hunk 1]
```diff
@@ -123,6 +123,25 @@ svg {
...
+// The default :focus-visible rendering differs from browser to
+// browser, changes over time, doesn't necessarily fit into the
+// Patternfly color scheme, looks ugly when used together with the
+// Patternfly :focus styling, and serves no purpose when :focus is
+// already visible.
+//
+// This would all be acceptable if these elements wouldn't receive
+// :focus-visible when focused with the mouse.
+//
+// https://github.com/patternfly/patternfly/issues/4121
```
Comment (KKoukiou) [inline pkg/lib/patternfly/patternfly-4-overrides.scss:139]: Removing the ring is very intrusive IMO, I would rather wait for the linked issue to get an answer!

**Comment by mvollmer** [comment]
If I would be king for a day, this is what I would do. :-)

Right now focusing a text input with the mouse looks like this:

Firefox:
image

Chrome:
image

GNOME Web:
image

The thick blue bottom border comes from Patternfly. Firefox adds a thin blue outline, Chrome adds a thick black outline, GNOME Web adds a faint dotted outline.

With the change in this PR, focussing a text input with the mouse looks like this:

Firefox:
image

Chrome:
image

GNOME Web:
image

GNOME Web doesn't seem to support :focus-visible but also has a suitably non-annoying focus ring.

**Comment by mvollmer** [comment]
This PR makes tabbing through a dialog visually less consistent on Chrome: some elements get styled according to Patternfly, some elements get the default thick black outline. It would be good to not interfere with :focus-visible styling for this reason, but the damage the default does to mouse-focused text inputs is just too high, IMO. Thus, if anything, we should probably take more control over :focus-visible and define it for all elements.

File: pkg/lib/patternfly/patternfly-4-overrides.scss (hunk 1 above)
Comment (mvollmer) [reply pkg/lib/patternfly/patternfly-4-overrides.scss:139]: Yes, but I have to get this out of my system. :-)

**Review by garrett — CHANGES_REQUESTED** [review]
While the "before" screenshots are all wildly inconsistent, the "after" screenshots do not look focused.

Chrome changed its focused outlines recently,
Firefox changed its default widgets in 89 (last week),
WebKit's is heavily influenced by macOS (and GTK's port might not reflect intent).

And, of course, all other browsers are basically Chrome.

But first rule of web development with accessibility in mind: Don't break focus states. (Often, this is in the form of `outline`.) The after ones do not look focused, therefore this needs work.

**Pull request closed without merging**